### PR TITLE
Create shipment form

### PR DESCRIPTION
**Pull request by tommysr**

### CODE REVIEW

File: app/src/components/Modals/Modal.svelte
Comment (tejks) [inline app/src/components/Modals/Modal.svelte:1]: I was thinking about something a little different. If works, don't change it. I will do this in the future.

File: app/src/components/ShipmentForm/DimensionsForm.svelte [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+<form use:form>
+	<div class="border-primary border-t text-primary-800 px-4 py-3 mb-5" role="alert">
+		<p class="font-bold">Action needed</p>
+		<p class="text-sm">Enter the dimensions of your shipment</p>
+	</div>
+
+	<DimensionsPick bind:isMetricTon />
+
+	{#each ['weight', 'width', 'height', 'depth', 'volume'] as name}
```
Comment (tejks) [inline app/src/components/ShipmentForm/DimensionsForm.svelte:92]: Separate variable looks better.

File: app/src/components/ShipmentForm/DetailsForm.svelte [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	import * as yup from 'yup';
+	import type { DetailsFormInterface } from './interfaces';
+
+	export let initialValues: DetailsFormInterface;
+	export let onSubmit;
+	export let onBack;
+
+	export let showModal = true;
+
+	const schema = yup.object({
```
Comment (tejks) [inline app/src/components/ShipmentForm/DetailsForm.svelte:16]: You can move all schemas to the schema folder. Makes the code clearer.

File: app/src/components/ShipmentForm/DetailsForm.svelte [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		count: yup.number().required(),
+		access: yup.string().transform(parseInt).required(),
+		priority: yup.number().required(),
+		fragility: yup.number().required()
+	});
+
+	const { form, data } = createForm<yup.InferType<typeof schema>>({
```
Comment (tejks) [inline app/src/components/ShipmentForm/DetailsForm.svelte:23]: Same with validation.

File: app/src/stores/orderForm.ts [hunk 4]
```diff
@@ -47,45 +53,98 @@ export function createFormStore() {
...
+			(registered && s.currentStage == FormStages.Price) ||
+			(!registered && s.currentStage == FormStages.Name)
+		) {
+			return false;
+		}
+
+		return true;
+	};
+
+	const checkStageValidity = (s: FormInterface) => {
```
Comment (tejks) [inline app/src/stores/orderForm.ts:79]: Isn't this data validated second time?

**Comment by tejks** [comment]
This FormStage looks really good.

File: app/src/stores/orderForm.ts (hunk 4 above)
Comment (tommysr) [reply app/src/stores/orderForm.ts:79]: it's not used, to be deleted entirely

File: app/src/components/ShipmentForm/DetailsForm.svelte (hunk 2 above)
Comment (tommysr) [reply app/src/components/ShipmentForm/DetailsForm.svelte]: moved

File: app/src/components/ShipmentForm/DetailsForm.svelte (hunk 3 above)
Comment (tommysr) [reply app/src/components/ShipmentForm/DetailsForm.svelte:17]: hmm, i would do it but i want to first resovle string/integer parsing, between felte library and dom, cause now theres too many strings

**Comment by matzayonc** [comment]
I'm just getting `Buffer` errors to be honest

**Comment by tejks** [comment]
> I'm just getting `Buffer` errors to be honest

Fixed

**Comment by tommysr** [comment]
Ready to merge.

File: app/src/components/ShipmentForm/DimensionsForm.svelte (hunk 1 above)
Comment (tommysr) [reply app/src/components/ShipmentForm/DimensionsForm.svelte:42]: later aligator

File: app/vite.config.ts [hunk 5]
```diff
@@ -1,10 +1,18 @@
...
+import { nodePolyfills } from 'vite-plugin-node-polyfills';
 import { defineConfig } from 'vitest/config';
 
 export default defineConfig({
-	plugins: [sveltekit()],
+	plugins: [
+		sveltekit(),
+		nodePolyfills({
+			globals: {
+				Buffer: true
```
Comment (matzayonc) [inline app/vite.config.ts:12]: <3

**Review by matzayonc — APPROVED** [review]

**Pull request merged**